lib/charms/layer/docker_registry.py

```diff
@@ -278,14 +279,20 @@ def _write_tls_blobs_to_files():
         ('tls-key-blob', 'tls-key-path')
     ]
 
-    for blob, path in blobs:
-        blob = charm_config.get(blob)
-        path = Path(charm_config.get(path))
+    for blob_key, path_key in blobs:
+        blob = charm_config.get(blob_key)
+        path = Path(charm_config.get(path_key))
 
-        if blob:
+        if blob and path:
             path.parent.mkdir(parents=True, exist_ok=True)
             path.write_bytes(base64.b64decode(blob))
 
+        elif not path:
+            hookenv.log('{} must be set.'.format(path_key))
+            return False
```

Comment:
Just log the fact that path is missing like you did.  IMO, there's no need to return anything here -- any callers should check for the existence of tls files without assuming this function wrote anything.  FWIW, we require that from callers because TLS may come from relation or config, so anything that cares about tls needs to check for the files, and not make assumptions that config-changed or relation-changed has written any data.